hollow/src/main/java/com/netflix/hollow/api/client/HollowClientUpdater.java

```diff
@@ -130,6 +134,13 @@ public synchronized boolean updateTo(long requestedVersion) throws Throwable {
         for (HollowConsumer.RefreshListener listener : localListeners)
             listener.refreshStarted(beforeVersion, requestedVersion);
 
+        for(HollowConsumer.RefreshListener listener : localListeners) {
+            // both pinning status of the version and headers map is required further
+            if (versionInfo.isPinned().isPresent() && versionInfo.getAnnouncementMetadata().isPresent()) {
+                listener.announcementDetected(requestedVersion, versionInfo.getAnnouncementMetadata().get(), versionInfo.isPinned().get());
```

Comment:
Sorry I realize we were going back and forth on whether this should be its own event or whether to collapse it into refreshStarted, I think given how it fires right around when refreshStarted fires, IMO it makes sense to add this as another flavor of refreshStarted, so
```
listener.refreshStarted(versionInfo, requestedVersion);
```
while also keeping the old one around as-is. Note we're also passing the versionInfo pojo instead of its constituent fields (client updater doesn't  care if versionInfo contains pin status and announcement metadata, the specific metrics listener does) .

Reply:
That makes sense based on how the events get fired in the listener. My concern is distinguishing between the two flavors of refreshStarted. Right now, refreshStarted instantiates metric builders and clears any existing state (header value maps in this case). just to clarify - is this more about renaming announcementDetected() method?